src/scenes/lobbyscreen.ts

```diff
@@ -0,0 +1,99 @@
+        super("LobbyScreen");
+        this.text = null;
+        // array of players
+        this.players = PlayerManager.getPlayers();
+        // sessionId, playername, playerId, role, turnIndex
+        this.me = SessionManager.getSessionInformation();
+    }
+
+    public syncPlayers = () => {
+        log("In syncPlayers");
+        this.players = PlayerManager.getPlayers();
+        this.update();
+    }
+
+    public syncMe = () => {
+        log("In syncMe")
+        this.me = SessionManager.getSessionInformation();
+        this.update()
+    }
+
+    public init(): void {
+        const playerManagerListener: UUID = PlayerManager.onSync.on(this.syncPlayers);
+        const sessionManagerListener: UUID = SessionManager.onSync.on(this.syncMe);
+
+        // on scene destroy free listener
+        this.events.on("destroy", () => {
+            PlayerManager.onSync.off(playerManagerListener);
+            SessionManager.onSync.off(sessionManagerListener);
+        });
+    }
+    
+    public preload(): void{
+        this.load.image("testButton", "assets/testButton.png");
+    }
+
+    public create(): void {
+        this.text = this.add
+            .text(350, 80, "Saboteur Lobby", {
+                fontFamily: "Goudy",
+                fontSize: 38,
+                color: "#ffffff",
+                stroke: "#000000",
+                strokeThickness: 8,
+                align: "center",
+            })
+            .setOrigin(0.5);
+        log(this, this.text);
+
+        const startButton: Phaser.GameObjects.Image = this.add.image(150, 540, "testButton");
+        startButton.setInteractive();
+        interactify(this, startButton, 1, () => { log("Click brown button"); });
+    }
+
+    private playerBackground(ypos: integer): void{
+        // Placeholder for the background on player display
+        this.graphics = this.add.graphics();
+        this.graphics.lineStyle(1, 0x000000, 1);
+        this.graphics.fillStyle(0x5C4033, 1.0);
+        // xpos, ypos, horizontal length, vertical length, corner radius
+        this.graphics.fillRoundedRect(100, ypos, 600, 27, 5);
+        this.graphics.strokeRoundedRect(100, ypos, 600, 27, 5);
+    }
+
+    public override update(): void {
+        for (let i: int = 0; i < this.players.length; i++){
+            let ypos: int = 150 + i*30;
+            let playername: string = this.players[i]
+            let fontstyle: string = "normal";
+            if (this.players[i] === this.me.playername){
+                playername = "You are: " + playername + ", Role: TBD";
+                fontstyle = "oblique";
+            }
+            this.playerBackground(ypos);
+            this.add.text(100, ypos, playername, { 
+                fontFamily: "Arial Black",
+                fontSize: 20,
+                color: "#e1e1e1",
+                stroke: "#000000",
+                strokeThickness: 3,
+                fontStyle: fontstyle,                
+            });
```

Comment:
it looks to me that every update all new texts and background get created and not properly destroyed, creating a memory leak.